deel/torchlip/utils/lconv_norm.py

```diff
@@ -24,24 +24,54 @@
 # rights reserved. DEEL is a research program operated by IVADO, IRT Saint Exupéry,
 # CRIAQ and ANITI - https://www.deel.ai/
 # =====================================================================================
-from typing import Tuple
+from typing import Tuple, Union
 
 import numpy as np
 import torch
 import torch.nn as nn
 import torch.nn.utils.parametrize as parametrize
 
 
+def compute_lconv_coef_1d(
+    kernel_size: Tuple[int],
+    input_shape: Tuple[int] = None,
+    strides: Tuple[int] = (1,),
+    padding_mode: str = "zeros",
+) -> float:
+    stride = strides[0]
+    k1 = kernel_size[0]
+
+    if (
+        (padding_mode in ["zeros", "same"])
+        and (stride == 1)
+        and (input_shape is not None)
+    ):
+        # See https://arxiv.org/abs/2006.06520
+        in_l = input_shape[-1]
+        k1_div2 = (k1 - 1) / 2
+        coefLip = in_l / (k1 * in_l - k1_div2 * (k1_div2 + 1))
+    else:
```

Comment:
this condition works for circular padding but probably not for "symmetric" or "reflect".

Reply:
Condition for applying the factor proposed in are the following:
(padding_mode in ["zeros"]) and (stride == 1) and (input_shape is not None)
Otherwise the factor is k_size/stride

As discussed, symmetric padding with RKO is k_size Lipschitz.
But reflect padding could be more thant k_size. We may reject this padding in SpectralConv layer ?

Reply:
Agreed